ch4/exercise4-2/printHash.go

```diff
@@ -0,0 +1,32 @@
+package main
+
+import (
+	"bufio"
+	"crypto/sha256"
+	"crypto/sha512"
+	"flag"
+	"fmt"
+	"os"
+)
+
+func main() {
+
+	var s384 = flag.Bool("sha384", false, "generate sha384 hash")
+	var s512 = flag.Bool("sha512", false, "generate sha512 hash")
```

Comment:
Nit: Multiple boolean flags allows multiple to be set.  E.g. `printHash --sha384 --sha512 foo` which allows for ambiguous intent and should probably be an error or documented behavior (in a "real" program)

IAs a less ambiguous alternative, 'd probably recommend a `flag.String()` and a case statement.